### PR TITLE
truvari: 2.1.1 -> 4.0.0

### PR DESCRIPTION
**Pull request by dotlambda**
Diff: https://github.com/spiralgenetics/truvari/compare/v2.1.1...v4.0.0

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cc @scalavision

### CODE REVIEW

**Comment by risicle** [comment]
Tests appear to hang @ `test_refine_one ran in 2 sec with 64/0 lines to STDERR/OUT`, both nixos x86_64 & macos 10.15.

**Comment by dotlambda** [comment]
> Tests appear to hang @ `test_refine_one ran in 2 sec with 64/0 lines to STDERR/OUT`, both nixos x86_64 & macos 10.15.

That's why it's marked as draft. I don't have enough time to look into this as I don't use the software but I thought the work I already did might be useful.

File: pkgs/applications/science/biology/truvari/default.nix [hunk 1]
```diff
@@ -1,42 +1,69 @@
...
 , python3Packages
+, runtimeShell
+, htslib
 }:
 
-python3Packages.buildPythonApplication rec {
+let
+  ssshtest = fetchFromGitHub {
+    owner = "ryanlayer";
+    repo = "ssshtest";
```
Comment (scalavision) [inline pkgs/applications/science/biology/truvari/default.nix:11]: what kind of owner / repo is this?

The original repository seems to have been removed. I don't use this tool anymore, maybe we should just remove it?

It seems to have moved here:

https://github.com/ACEnglish/truvari

but I can not confirm that.

**Review by scalavision — CHANGES_REQUESTED** [review]

File: pkgs/applications/science/biology/truvari/default.nix (hunk 1 above)
Comment (dotlambda) [reply pkgs/applications/science/biology/truvari/default.nix:11]: That's just a bash script used to run tests.

Comment (scalavision) [reply pkgs/applications/science/biology/truvari/default.nix:11]: Thanks, really nice work! 😊

File: pkgs/applications/science/biology/truvari/default.nix [hunk 2]
```diff
@@ -1,42 +1,69 @@
...
-    acebinf
     bwapy
-    joblib
     pandas
   ];
 
-  # no tests
-  doCheck = false;
+  makeWrapperArgs = [
+    "--prefix" "PATH" ":" (lib.makeBinPath [ htslib ])
```
Comment (natsukium) [inline pkgs/applications/science/biology/truvari/default.nix:47]:
```suggestion
    "--prefix" "PATH" ":" (lib.makeBinPath [ htslib bcftools ])
```

File: pkgs/applications/science/biology/truvari/default.nix [hunk 3]
```diff
@@ -1,42 +1,69 @@
...
+  ];
+
   pythonImportsCheck = [ "truvari" ];
 
+  nativeCheckInputs = [
+    htslib
+  ] ++ (with python3Packages; [
+    coverage
+  ]);
```
Comment (natsukium) [inline pkgs/applications/science/biology/truvari/default.nix:56]:
```suggestion
  nativeCheckInputs = [
    htslib
    bcftools
  ] ++ (with python3Packages; [
    anybadge
    coverage
  ]);
```

**Review by natsukium — COMMENTED** [review]
The lack of `bcftools` seems to cause the `test_refine_one` to hang.
In my diff, all tests pass correctly.

**Comment by dotlambda** [comment]
@natsukium Would you like to become a maintainer for truvari?

File: pkgs/applications/science/biology/truvari/default.nix (hunk 3 above)
Comment (dotlambda) [reply pkgs/applications/science/biology/truvari/default.nix]: Works without `anybadge` for me.

**Comment by natsukium** [comment]
Alright, please add me too.

**Review by natsukium — APPROVED** [review]

**Review by natsukium — APPROVED** [review]

**Pull request merged**